src/tezos_interop/tezos_interop.mli

```diff
@@ -34,7 +35,7 @@ module Consensus : sig
     transactions : transaction list;
   }
   val listen_operations : t -> on_operation:(operation -> unit) -> unit
-  val fetch_validators : t -> (Key_hash.t list, string) result Lwt.t
+  val fetch_validators : t -> ((Key_hash.t * Uri.t) list, string) result Lwt.t
```

Comment:
Question: should the type of this be `Uri.t option` instead of `Uri.t`? Supposing there's a single key hash that doesn't have an entry in discovery.mligo - do we want to fail the entire node for that?

Reply:
True, I guess it should be the same as having a node that doesn't respond. But I think that should go in a different PR

Reply:
@anchpop this should not go in a different PR, this is an unhandled branch.